_posts/2019-10-20-learning-go-resources.md

```diff
@@ -0,0 +1,67 @@
+---
+title: "Some resources for learning Go"
+published: 2019-10-20 13-01-02 +01:00
+tags: programming go
+---
+
+I'm starting on a Go project for the first time in a while, and a few people
+have asked me how to starting writing Go. The last time I spent a significant
+time writing Go was over a year ago, so while I also needed to refresh my
+knowledge, there have been a few new things released since.
+
+I learn best in two phases: first by reading a lot about a subject, and then
+building something. The project itself is the last bit, but going into it, I
+want to be as primed as I can to hit the ground running. That all said, _what_
+I read is essential. I don't find programming books which walk through things
+like type primitives in great depth remotely engaging &mdash; I want to know
+just enough to be productive.
+
+I'm assuming here that you're coming from the position of already being
+confident with a programming language and testing. I don't think I'd start with
+learning Go from the beginning.
+
+## [Learn Go with Tests](https://quii.gitbook.io/learn-go-with-tests/)
+
+_Learn Go with Tests_ provides a lovely introduction to the basics of Go while
+taking you through test-driving what you're learning. This is the first time
+I've seen this approach, and I've found it's helped me get back into the
+language itself.
+
+## [Go in Practice](http://goinpracticebook.com)
+
+Once you've got an understanding of the basics of Go and have written a little,
+I'd recommend _Go in Practice_. It'll run through some common things you'll
+need to do and expose you to some patterns as you go. You might well find you
+can use the examples mostly as-is, which I find helps in making it feel like
+you're making good progress in building something.
+
+I read this cover to cover, but you don't need to do that. I find once I have,
+though, it's much easier for me to pick the book up and refer to something I
+then need.
+
+## [Ultimate Go Programming Course]()
+
+
+Once I've built a few things, I then want to go into some more depth. I find
```

Comment:
>Once I've built a few **simple projects** ~~things~~, I then want to go into some more depth **[explain]**.

e.g.

>I then want to go into some more depth **on how to structure more complex applications**

Reply:
Nice. I think this is something I'm likely to try and write up next. Maybe a tour of how to structure Go projects.